### PR TITLE
Redesigned the UI slightly

### PR DESCRIPTION
**Pull request by justjanne**
I redesigned some parts of the UI, including the drawers.

### CODE REVIEW

File: QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java [hunk 1]
```diff
@@ -581,150 +573,136 @@ void setPanelFragment(int side, Fragment fragment) {
...
+                        actionbar.getWrappedToolbar(), /* nav drawer icon to replace 'Up' caret */
+                        R.string.hint_drawer_open, /* "open drawer" description */
+                        R.string.hint_drawer_close /* "close drawer" description */
+                ) {
+                    /**
+                     * Called when a drawer has settled in a completely closed state.
+                     */
+                    public void onDrawerClosed(View drawerView) {
+                        updateBufferRead();
+                        updateBufferRead();
```
Comment (kenborge) [inline QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java]: Duplicate code

File: QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java [hunk 2]
```diff
@@ -581,150 +573,136 @@ void setPanelFragment(int side, Fragment fragment) {
...
+                     */
+                    public void onDrawerOpened(View drawerView) {
+                        manager.closeDrawer(Side.RIGHT);
+                        setTitleAndMenu();
+                    }
+
+                    public void onDrawerSlide(View drawerView, float slideOffset) {
+                        if (manager.isDrawerVisible(Side.RIGHT))
+                            manager.closeDrawer(Side.RIGHT);
+                        super.onDrawerSlide(drawerView,0);
```
Comment (kenborge) [inline QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java]: It this correct? Shouldn't is just pass the slideOffset?

File: QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java [hunk 3]
```diff
@@ -581,150 +573,136 @@ void setPanelFragment(int side, Fragment fragment) {
...
         }
 
-        public void closeDrawerOpposite(View drawerView) {
-            if (hasDrawer && drawerView.getTag()=="LEFT")
-                closeDrawer(Gravity.START);
-            else if (hasDrawer)
-                closeDrawer(Gravity.END);
-        }
+        private void closeDrawer(Side side) {
+            if (side==Side.RIGHT||side==Side.BOTH) {
```
Comment (kenborge) [inline QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java]: The left drawer closer has a null check right below, does the right side not need that?

File: QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java [hunk 4]
```diff
@@ -581,150 +573,136 @@ void setPanelFragment(int side, Fragment fragment) {
...
-            setPanelFragment(Gravity.END, null);
-            setPanelFragment(Gravity.NO_GRAVITY, null);
 
-            bufferFragment = null;
-            nickFragment = null;
-            chatFragment = null;
-            detailFragment = null;
-        }
+        public void openDrawer(Side side) {
+            if (side==Side.RIGHT) {
```
Comment (kenborge) [inline QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java]: Same question here, no null check is that correct?

File: QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java (hunk 2 above)
Comment (justjanne) [inline QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java]: The guidelines say that, if the drawer slides over the button, the button always should stay in position 0.

File: QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java (hunk 3 above)
Comment (justjanne) [inline QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java]: The right one always exists as drawer, while the left one sometimes exists as drawer (on phones) and sometimes is always-open (on tablets)

File: QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java (hunk 4 above)
Comment (justjanne) [inline QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java]: And same answer here, the right drawer exists in every layout as a drawer.

**Comment by kenborge** [comment]
Remove the duplicate code, and move the menu back and this one looks good

**Comment by justjanne** [comment]
should both be done now, and more additions as well.

**Comment by kenborge** [comment]
One issue i noticed though is that the buffer drawer is open when you first connect, so you don't see the loading screen that is happening in the main view

**Comment by justjanne** [comment]
This issue is already been worked on, it was just necessary to do it this way first to do it properly later.

**Comment by justjanne** [comment]
Lol, I rewrote half of the opening/closing logic and the result is exactly the same as before?

Anyway, I think the bug is fixed now.

**Pull request merged**